### PR TITLE
CXP-650: Move the Connection dashboard to the Connect space

### PR DESCRIPTION
**Pull request by tomglvng**
Capture du 2021-04-22 15-55-32

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/front/src/connect/pages/RedirectConnectionDashboardToConnectMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import {ClientErrorIllustration, Breadcrumb} from 'akeneo-design-system';
+import React from 'react';
+import {Translate, useTranslate} from '../../shared/translate';
+import {HelperLink, PageHeader} from '../../common';
+import {UserButtons} from '../../shared/user';
+import styled from 'styled-components';
+import {Heading} from '../../common/components/EmptyState';
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/connect/pages/RedirectConnectionDashboardToConnectMenu.tsx:7]:
```suggestion
import {EmptyState, HelperLink, PageHeader} from '../../common';
import {UserButtons} from '../../shared/user';
import styled from 'styled-components';
```

File: src/Akeneo/Connectivity/Connection/front/src/connect/pages/RedirectConnectionDashboardToConnectMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    return (
+        <>
+            <PageHeader breadcrumb={breadcrumb} userButtons={<UserButtons />}>
+                {translate('pim_menu.item.connection_audit')}
+            </PageHeader>
+
+            <PageContent>
+                <ClientErrorIllustration width={500} height={250} />
+
+                <Heading>{translate('akeneo_connectivity.connection.connect.redirect.title')}</Heading>
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/connect/pages/RedirectConnectionDashboardToConnectMenu.tsx:38]:
```suggestion
                <EmptyState.Heading>{translate('akeneo_connectivity.connection.connect.redirect.title')}</EmptyState.Heading>
```

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/RedirectConnectionDashboardToConnectMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, {StrictMode} from 'react';
+import {AkeneoThemeProvider} from './akeneo-theme-provider';
+import {withDependencies} from './dependencies-provider';
+import {RedirectConnectionDashboardToConnectMenu as RedirectConnectionDashboardToConnectMenuPage} from '../connect/pages/RedirectConnectionDashboardToConnectMenu';
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/infrastructure/RedirectConnectionDashboardToConnectMenu.tsx:4]: I'm not sure that the import rename is useful here

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/dependencies-provider.tsx [hunk 4]
```diff
@@ -35,8 +35,8 @@ const DependenciesProvider = ({children, ...dependencies}: PropsWithChildren<Pro
     </RouterContext.Provider>
 );
 
-export const withDependencies = (Component: ElementType) => {
-    return ({dependencies, ...props}: {dependencies: Props}) => (
+export const withDependencies = <AdditionalProps, >(Component: ElementType) => {
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/infrastructure/dependencies-provider.tsx:38]: I don't understand the addition of the generic parameter `AdditionalProps` here?

**Review by pchasle — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/RedirectConnectionDashboardToConnectMenu.tsx (hunk 3 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/front/src/infrastructure/RedirectConnectionDashboardToConnectMenu.tsx:4]: conflict with the current component name.

Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/src/infrastructure/RedirectConnectionDashboardToConnectMenu.tsx:4]: Ah yes, my bad

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/dependencies-provider.tsx (hunk 4 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/front/src/infrastructure/dependencies-provider.tsx:38]: You are right, we should discuss it in another PR, but there was an interesting issue with `withDependencies`.
It does support unknown props, see https://github.com/akeneo/pim-community-dev/pull/14293/files/bbb87c78ebdf7f5c0b20d8c276f0229d561b2d48#diff-45c83e4eebccd662c1fc1ae61774124ac3ff01bb06ec14085ee40b9a6f5d69b5R41
But there is no way to define those in the props !
By using it as a Generic like this, you can do:
```
withDependencies<{foo: string}>(...)
```

Comment (tomglvng) [reply src/Akeneo/Connectivity/Connection/front/src/infrastructure/dependencies-provider.tsx]: Rollback

**Review by tseho — APPROVED** [review]

**Pull request closed without merging**